File: cli/src/main/java/de/mobanisto/lanchat/SendFile.java
package de.mobanisto.lanchat;

import java.io.IOException;
import java.io.OutputStream;
import java.net.Socket;
import java.net.UnknownHostException;
import java.nio.channels.Channels;
import java.nio.channels.FileChannel;
import java.nio.channels.WritableByteChannel;
import java.nio.file.Path;
import java.nio.file.Paths;

public class SendFile
{

	public static void main(String[] args)
			throws UnknownHostException, IOException
	{
		if (args.length != 2) {
			System.out.print("usage: send-file <ip> <file>");
			System.exit(1);
		}

		String argIp = args[0];
		String argInput = args[1];
		Path input = Paths.get(argInput);

		System.out.println("Connecting...");
		try (Socket socket = new Socket(argIp, 5001)) {
			System.out.println("Sending file...");
			OutputStream os = socket.getOutputStream();
			WritableByteChannel channel = Channels.newChannel(os);
			FileChannel inputChannel = FileChannel.open(input);
			long transferred = inputChannel.transferTo(0, Integer.MAX_VALUE,
					channel);
			System.out.println(String.format("Sent %d bytes", transferred));
		}
	}

}

File: desktop/src/main/kotlin/de/mobanisto/lanchat/SendFile.kt
package de.mobanisto.lanchat

import com.github.ajalt.clikt.core.CliktCommand
import com.github.ajalt.clikt.parameters.arguments.argument
import com.github.ajalt.clikt.parameters.options.default
import com.github.ajalt.clikt.parameters.options.option
import com.github.ajalt.clikt.parameters.types.int
import com.github.ajalt.clikt.parameters.types.path
import java.io.OutputStream
import java.net.Socket
import java.nio.channels.Channels
import java.nio.channels.FileChannel

class SendFile : CliktCommand(name = "send-file") {

    private val port: Int by option(
        "--port",
        help = "Port to send files on"
    ).int().default(5001)

    private val ip by argument()
    private val input by argument().path(mustExist = true)

    override fun run() {
        println("Connecting...")
        Socket(ip, port).use { socket ->
            println("Sending file...")
            val os: OutputStream = socket.getOutputStream()
            val channel = Channels.newChannel(os)
            val inputChannel = FileChannel.open(input)
            val transferred = inputChannel.transferTo(0, Int.MAX_VALUE.toLong(), channel)
            println(String.format("Sent %d bytes", transferred))
        }
    }
}
